src/protocol.cpp

```diff
@@ -199,3 +199,127 @@ const std::vector<std::string> &getAllNetMessageTypes()
 {
     return allNetMessageTypesVec;
 }
+
+
+uint8_t GetShortCommandIDFromCommand(const std::string cmd)
+{
+    if (cmd == NetMsgType::ADDR) {
+        return NetMsgType::ADDR_SHORT_ID;
+    } else if (cmd == NetMsgType::BLOCK) {
+        return NetMsgType::BLOCK_SHORT_ID;
+    } else if (cmd == NetMsgType::BLOCKTXN) {
+        return NetMsgType::BLOCKTXN_SHORT_ID;
+    } else if (cmd == NetMsgType::CMPCTBLOCK) {
+        return NetMsgType::CMPCTBLOCK_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::FEEFILTER) {
+        return NetMsgType::FEEFILTER_SHORT_ID;
+    } else if (cmd == NetMsgType::FILTERADD) {
+        return NetMsgType::FILTERADD_SHORT_ID;
+    } else if (cmd == NetMsgType::FILTERCLEAR) {
+        return NetMsgType::FILTERCLEAR_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::FILTERLOAD) {
+        return NetMsgType::FILTERLOAD_SHORT_ID;
+    } else if (cmd == NetMsgType::GETADDR) {
+        return NetMsgType::GETADDR_SHORT_ID;
+    } else if (cmd == NetMsgType::GETBLOCKS) {
+        return NetMsgType::GETBLOCKS_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::GETBLOCKTXN) {
+        return NetMsgType::GETBLOCKTXN_SHORT_ID;
+    } else if (cmd == NetMsgType::GETDATA) {
+        return NetMsgType::GETDATA_SHORT_ID;
+    } else if (cmd == NetMsgType::GETHEADERS) {
+        return NetMsgType::GETHEADERS_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::HEADERS) {
+        return NetMsgType::HEADERS_SHORT_ID;
+    } else if (cmd == NetMsgType::INV) {
+        return NetMsgType::INV_SHORT_ID;
+    } else if (cmd == NetMsgType::MEMPOOL) {
+        return NetMsgType::MEMPOOL_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::MERKLEBLOCK) {
+        return NetMsgType::MERKLEBLOCK_SHORT_ID;
+    } else if (cmd == NetMsgType::NOTFOUND) {
+        return NetMsgType::NOTFOUND_SHORT_ID;
+    } else if (cmd == NetMsgType::PING) {
+        return NetMsgType::PING_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::PONG) {
+        return NetMsgType::PONG_SHORT_ID;
+    } else if (cmd == NetMsgType::SENDCMPCT) {
+        return NetMsgType::SENDCMPCT_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::SENDHEADERS) {
+        return NetMsgType::SENDHEADERS_SHORT_ID;
+        ;
+    } else if (cmd == NetMsgType::TX) {
+        return NetMsgType::TX_SHORT_ID;
+    } else if (cmd == NetMsgType::VERACK) {
+        return NetMsgType::VERACK_SHORT_ID;
+    } else if (cmd == NetMsgType::VERSION) {
+        return NetMsgType::VERSION_SHORT_ID;
+        ;
+    }
+    return 0; //no short command
+}
+
+bool GetCommandFromShortCommandID(uint8_t shortID, std::string& cmd)
+{
+    if (shortID == NetMsgType::ADDR_SHORT_ID) {
```

Comment:
Have you consider a `switch` statement as an alternative?

Reply:
Can you elaborate on the differences? Aren't the compiler differences marginal?

Reply:
> Can you elaborate on the differences? Aren't the compiler differences marginal?

It does seem like a switch statement might be a good alternative to the `if...else ifs` here, to make explicit the nature of the operation (testing a single value against a set of scoped enumerations or array of values) and possibly generate better code with a jump table instead of repeatedly checking individual values.

A translation table could be an alternative as well.

Reply:
I agree that a switch statement is easier to read(and see what's missing if we'll have more `NetMsgType` types)

Reply:
Seems to be a matter of taste. I'd like to keep it as is.

Reply:
I think the advantage of a (hash)table is that it could be built in one central function, making the mapping in both directions, which leaves only one place to update for new message types (and making sure they never go out of sync).